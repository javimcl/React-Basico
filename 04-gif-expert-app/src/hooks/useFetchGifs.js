import { useState } from "react";
import { useEffect } from "react/cjs/react.development";
import { getGifs } from "../helpers/getGifs";



export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {

        getGifs(category)
            .then(imgs => {
               // setTimeout(() => {
                    setState({
                        data: imgs,
                        loading: false
                    });
               // }, 3000);
            })

    }, [category])

    // setTimeout(() => {
    //     setState({
    //         data: [1,2,3,4,5],
    //         loading:false
    //     })
    // }, 3000);

    return state; //{data:[], loading:true}
}
