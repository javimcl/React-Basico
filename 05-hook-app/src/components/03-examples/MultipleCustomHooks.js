

import React from 'react'
import { useCounter } from '../../hooks/useCounter'
import { useFetch } from '../../hooks/useFetch'
import '../02-useEffect/effect.css'

export const MultipleCustomHooks = () => {

    const {counter,increment} = useCounter(1);

    const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
    console.log(data);
    const {author, quote} = !!data && data[0];

    console.log(author, quote);
    return (
        <div>
            <h1>Breaking bas Quotes</h1>
            <hr />
            {
                loading
                    ?
                    (
                        <div className='alert alert-info text-center'>
                            Loading....
                        </div>
                    )
                    :
                    (
                        <blockquote className='blockquote text-right'>
                            <p className='mb-0'>{quote}</p>
                            <br/>

                            <footer className='blockquote-footer' >{author}</footer>
                        </blockquote>
                    )
            }

            <button 
            className='btn btn-primary'
            onClick={increment}>
                Siguiente
            </button>


        </div>
    )
}
