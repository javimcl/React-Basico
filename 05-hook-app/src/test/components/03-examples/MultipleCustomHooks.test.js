import { render } from "@testing-library/react";
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'
import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks";
import { useFetch } from "../../../hooks/useFetch";
import { useCounter } from "../../../hooks/useCounter";
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Pruebas en <MultipleCustom/>', () => {

    beforeEach( () => {
    useCounter.mockReturnValue({
        counter: 10,
        increment: () => { }
    })
})

    test('debe de mostrarse correctamente', () => {


        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })
        const wrapper = shallow(<MultipleCustomHooks />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar la informacion', () => {

        useFetch.mockReturnValue({
            data: [{
                author: 'Javier',
                quote: 'Hola Mundo'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);

        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('Hola Mundo');
        expect(wrapper.find('footer').text().trim()).toBe('Javier');
console.log(wrapper);
    });


});
