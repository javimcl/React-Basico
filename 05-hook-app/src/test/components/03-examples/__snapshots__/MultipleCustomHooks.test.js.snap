// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustom/> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Breaking bas Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading....
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente
  </button>
</div>
`;
